client/src/components/Visualization/charts/Geochart.js

```diff
@@ -1,50 +1,76 @@
-/**
+/** 
  * Description: Geochart implements a Google geochart component.
  * Author: Kira Toal
  * Date: 2020/07/13
  */
 
-import { Chart } from 'react-google-charts';
-import React from 'react';
 import { ads } from '../../../firebase/FirestoreDocumentReader';
+import { Chart } from "react-google-charts";
 import firebase from '../../../firebase/firebase';
+import React, { useState, useEffect } from 'react';
 import { states } from './StateDataParser';
+import { app, database } from '../../../firebase/firebase';
 
 const Geochart = () => {
-  /*
-   * Currently, getData assigns a meaningless random number to each state.
-   * In the future, once the states dictionary contains ad information, getData will
-   *    retrieve information such as minimum ad impressions and maximum ad spend.
-   */
-
-  function getData() {
-    const data = [['State', 'Random Number']];
-    for (const state in states) {
-      data.push([state, Math.floor(Math.random() * Math.floor(1000))]);
+
+  const [adTotal, setAdTotal] = useState(0);
+
+  class GeochartAd {
+    constructor(id, impressionsMin) {
+    /**
+    * @param {string} id
+    * @param {long} impressionsMin
+    */
+      this.id = id;
+      this.impressionsMin = impressionsMin;
+    }
+    toString() {
+      return this.id + ', ' + this.impressionsMin;
+    }
+  }
+
+  // Firestore data converter converts snapshots to custom objects.
+  const adConverter = {
+    toFirestore: function(ad) {
```

Comment:
might be shorter to do: 
` const { id, impressionMin } = ad;` 
then
`return { id, impressionMin };`